packages/block-editor/src/components/block-list/use-block-props/use-zoom-out-mode-exit.js

```diff
@@ -16,29 +16,30 @@ import { unlock } from '../../../lock-unlock';
  * @param {string} clientId Block client ID.
  */
 export function useZoomOutModeExit( { editorMode } ) {
-	const { getSettings } = useSelect( blockEditorStore );
-	const { __unstableSetEditorMode } = unlock(
+	const { getEditorMode } = useSelect( ( select ) => {
+		const { __unstableGetEditorMode } = select( blockEditorStore );
+		return {
+			getEditorMode: __unstableGetEditorMode,
+		};
+	}, [] );
+
+	const { __unstableSetEditorMode, setZoomOut } = unlock(
 		useDispatch( blockEditorStore )
 	);
 
 	return useRefEffect(
 		( node ) => {
-			if ( editorMode !== 'zoom-out' ) {
+			if ( editorMode !== 'compose' ) {
 				return;
 			}
 
 			function onDoubleClick( event ) {
 				if ( ! event.defaultPrevented ) {
 					event.preventDefault();
-
-					const { __experimentalSetIsInserterOpened } = getSettings();
-
-					if (
-						typeof __experimentalSetIsInserterOpened === 'function'
-					) {
-						__experimentalSetIsInserterOpened( false );
+					if ( getEditorMode() === 'compose' ) {
```

Comment:
Do we also want to check for being zoomed out here?

Reply:
If you are zoomed out but _not_ in "compose" mode do we want double click to zoom in to work?